tests/server/routes.js

```diff
@@ -214,13 +216,19 @@ define([
   // speed up the tests.
   var checkedUrlPromises = {};
 
-  function checkUrls(urls) {
+  function checkUrls(origin, urls) {
     var requests = urls.map(function (url) {
       if (checkedUrlPromises[url]) {
         return checkedUrlPromises[url];
       }
 
-      var promise = makeRequest(url, {})
+      var requestOptions = {
```

Comment:
My comment is the same as @jbuck's, perhaps we should call `isExternalSite` and only add the `Origin` header if making a request to a 3rd party.